ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install_rocm.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+set -ex
+
+#move to rocm package
+sudo ./amdgpu-install -y --usecase=graphics,rocm
+
+
+#Add self to render and video groups so they can access gpus.
+sudo usermod -a -G render $(logname)
+sudo usermod -a -G video $(logname)
+
+#add future new users to the render and video groups.
+echo 'ADD_EXTRA_GROUPS=1' | sudo tee -a /etc/adduser.conf
+echo 'EXTRA_GROUPS=video' | sudo tee -a /etc/adduser.conf
+echo 'EXTRA_GROUPS=render' | sudo tee -a /etc/adduser.conf
+
+#add nofile limits
+string_so="*               soft    nofile          1048576"
+line=$(cat /etc/security/limits.conf | grep "soft    nofile")
+cat /etc/security/limits.conf | sed -e "s/$line/$string_so/" > temp_file.txt
+sudo mv temp_file.txt /etc/security/limits.conf
+string_ha="*               hard    nofile          1048576"
+line=$(cat /etc/security/limits.conf | grep "hard    nofile")
+cat /etc/security/limits.conf | sed -e "s/$line/$string_ha/" > temp_file.txt
+sudo mv temp_file.txt /etc/security/limits.conf
+
+cat /etc/security/limits.conf | grep -v "  stack" > tmplimits.conf
+sudo mv tmplimits.conf /etc/security/limits.conf
+
+echo blacklist amdgpu | sudo tee -a /etc/modprobe.d/blacklist.conf
+sudo update-initramfs -c -k $(uname -r)
+
+echo "Writing gpu mode probe in init.d"
```

Comment:
Can you use a heredoc instead of individual echo cmds?

Reply:
done